torch/utils/data/datapipes/iter/routeddecoder.py

```diff
@@ -1,38 +1,47 @@
-from torch.utils.data import IterDataPipe
+import re
+
+from io import BufferedIOBase
+from typing import Any, Callable, Iterable, Iterator, Sized, Tuple
+
+from torch.utils.data import IterDataPipe, functional_datapipe
 from torch.utils.data.datapipes.utils.decoder import (
     Decoder,
     basichandlers as decoder_basichandlers,
     imagehandler as decoder_imagehandler)
 
-from typing import Iterable, Iterator, Union, List, Tuple, Any, Callable
-from io import BufferedIOBase
 
+# Extract extension from pathname
+def _default_key_fn(pathname):
+    return re.sub(r".*[.]", "", pathname)
```

Comment:
What is the advantage over `os.path.splitext(pathname)[1]`?

Reply:
Advantage of `re.sub(r".*[.]", "", pathname)`:
1. Directly output extension without `.`
2. For `.temp`, it returns `temp` rather than empty string. (Not sure if empty string should be expected as a key)

Advantage of `os.path.splitext(pathname)[1]`
1. Can take pathlike object

Reply:
Second thought, maybe it's not advantage for `.temp` file. It's just a hidden file without extension. 
I will change it into `os.path.splitext` then.